### PR TITLE
BET-4337: Upgrade maas provider CRD to v1beta1

### PR DESCRIPTION
**Pull request by AmitSahastra**
Upgrade maas provider CRD to v1beta1

- [x] Tested v1beta1 mass cluster creation using clusterctl 1.1.2
```
 amit@Amits-MacBook-Pro  ~/Spectro/sourcecode/my-cluster-api-provider-maas/cluster-api-provider-maas   betav1 ±✚  clusterctl_1.1 describe cluster t-cluster --grouping --show-conditions all
NAME                                                          READY  SEVERITY  REASON  SINCE  MESSAGE
Cluster/t-cluster                                             True                     17m
│           ├─ControlPlaneInitialized                         True                     17m
│           ├─ControlPlaneReady                               True                     17m
│           └─InfrastructureReady                             True                     17m
├─ClusterInfrastructure - MaasCluster/t-cluster               True                     17m
│             ├─APIServerAvailable                            True                     17m
│             └─LoadBalancerReady                             True                     21m
├─ControlPlane - KubeadmControlPlane/t-cluster-control-plane  True                     17m
│ │           ├─Available                                     True                     17m
│ │           ├─CertificatesAvailable                         True                     21m
│ │           ├─ControlPlaneComponentsHealthy                 True                     17m
│ │           ├─EtcdClusterHealthy                            True                     17m
│ │           ├─MachinesCreated                               True                     21m
│ │           ├─MachinesReady                                 True                     17m
│ │           └─Resized                                       True                     17m
│ └─Machine/t-cluster-control-plane-9g4r8                     True                     17m
│               ├─APIServerPodHealthy                         True                     17m
│               ├─BootstrapReady                              True                     21m
│               ├─ControllerManagerPodHealthy                 True                     16m
│               ├─EtcdMemberHealthy                           True                     17m
│               ├─EtcdPodHealthy                              True                     16m
│               ├─InfrastructureReady                         True                     17m
│               ├─NodeHealthy                                 True                     4m19s
│               └─SchedulerPodHealthy                         True                     16m
└─Workers
  └─MachineDeployment/t-cluster-md-0                          True                     4m18s
    │           └─Available                                   True                     4m18s
    └─Machine/t-cluster-md-0-cd59678f5-ls4rd                  True                     13m
                  ├─BootstrapReady                            True                     17m
                  ├─InfrastructureReady                       True                     13m
                  └─NodeHealthy                               True                     4m18s

New clusterctl version available: v1.1.2 -> v1.1.3
https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.3

 ✘ amit@Amits-MacBook-Pro  ~/Spectro/sourcecode/my-cluster-api-provider-maas/cluster-api-provider-maas   betav1 ±✚  clusterctl_1.1  move --to-kubeconfig=/Users/amit/.maaskubeconfig/config
Performing move...
Discovering Cluster API objects
Moving Cluster API objects Clusters=1
Moving Cluster API objects ClusterClasses=0
Creating objects in the target cluster
Deleting objects from the source cluster

New clusterctl version available: v1.1.2 -> v1.1.3
https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.1.
```

- [x] Tested v1alpha3 mass cluster creation using clusterctl 1.1.2

### CODE REVIEW

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#" title="Author self-approved">AmitSahastra</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jzhoucliqr**
You can assign the PR to them by writing `/assign @jzhoucliqr` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
@AmitSahastra: The following test **failed**, say `/retest` to rerun all failed tests:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
cluster-api-provider-maas_build | 62f398c73afb9c0edeaa4a01a0c973163b632f87 | [link](http://34.120.121.77/view/gcs/capi-prow-artifacts/pr-logs/pull/spectrocloud_cluster-api-provider-maas/37/cluster-api-provider-maas_build/1505789414859608064) | `/test cluster-api-provider-maas_build`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

File: pkg/maas/scope/machine.go
Comment (sadysnaat) [inline pkg/maas/scope/machine.go]: let's remove this

File: main.go
Comment (sadysnaat) [inline main.go:121]: let's create a separate log like before WithName("remote").WithName("ClusterCacheTracker")

if we don't want to use ctrl.log we can use 
log.WithName("remote").WithName("ClusterCacheTracker")

**Review by sadysnaat — DISMISSED** [review]

File: pkg/maas/scope/machine.go
Comment (AmitSahastra) [reply pkg/maas/scope/machine.go]: Commented lines are removed

File: main.go
Comment (AmitSahastra) [reply main.go:121]: Fixed

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#" title="Author self-approved">AmitSahastra</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#issuecomment-1078200378" title="Approved">sadysnaat</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jzhoucliqr**
You can assign the PR to them by writing `/assign @jzhoucliqr` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#" title="Author self-approved">AmitSahastra</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#issuecomment-1078200378" title="Approved">sadysnaat</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#" title="Author self-approved">AmitSahastra</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#issuecomment-1080424994" title="Approved">sadysnaat</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by sadysnaat — APPROVED** [review]

**Comment by spectro-prow** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#" title="Author self-approved">AmitSahastra</a>*, *<a href="https://github.com/spectrocloud/cluster-api-provider-maas/pull/37#pullrequestreview-922907641" title="Approved">sadysnaat</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=spectrocloud%2Fcluster-api-provider-maas).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/spectrocloud/cluster-api-provider-maas/blob/main/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
